### PR TITLE
Confirm extension in use in sb2 serialization

### PR DESCRIPTION
**Pull request by ktbee**
### Resolves

A few related issues: 
- https://github.com/LLK/scratch-vm/issues/1433 
- https://github.com/LLK/scratch-vm/issues/1635 
- https://github.com/LLK/scratch-vm/issues/1062

### Proposed Changes

This PR checks if there is a discrepancy between the stage target's `videoOn` and `videoTransparency` properties and what the video currently is displaying, and updates those properties accordingly. 

### Reason for Changes

Currently there is a race condition when extensions are loaded and use `this.runtime.getTargetForStage`. Targets are installed after the extensions are loaded, so the stage target's properties aren't available to load until after the targets are updated. This PR checks for discrepancies whenever targets are updated, and updates the project's video properties if they are different. I made an issue to handle this a little more neatly by adding an event for when targets are installed (https://github.com/LLK/scratch-vm/issues/1639), but checking when targets are updated could be a solution in the meantime.

### CODE REVIEW

**Review by fsih — CHANGES_REQUESTED** [review]
A couple issues
1. We would strongly prefer if the Video Sensing category didn't load into the project at all when no blocks from it are present. (It appears at the bottom of the list of block categories.) It's a particularly sketchy extension to have appear when you didn't expect it 
image

2. When the video sensing extension is added by the user, the video should show up on stage, but after this change it does not.
To repro click here
image
then here
image

@kchadha suggests that it might be a good solution to ignore loading a monitor from an extension category that isn't already loaded.

**Comment by ktbee** [comment]
Thanks for taking a look, @fsih. That feedback all makes sense, I'll make those updates.

**Comment by ktbee** [comment]
By the way @fsih, do you have any example projects where you're seeing the video sensing extension is added to the bottom of the list of categories or where you can't add it to the stage? I haven't been able to repro that yet, for example Bryce's project demoing the original issue isn't adding the video extension for me at the bottom of the block categories (238455324).

**Comment by fsih** [comment]
@ktbee that's odd, I got it on all the projects listed in https://github.com/LLK/scratch-vm/issues/1433 . I'm on Mac Chrome 69 in case that's relevant. Here's what I see
extensionload

**Comment by fsih** [comment]
@ktbee Here is what I see for the video loading (will pull vs develop)
videoload2

**Comment by ktbee** [comment]
@fsih thanks for the clarification and the screencasts, very helpful! I've simplified this PR after some offline suggestions from @kchadha as well. Currently it just addresses #1433 by checking that only extensions used in blocks or visible monitors are added to the set of `extensionIDs` for a serialized sb2 project.  

In this version I confirmed that the video shows on the stage when you add the video extension. Also the video extensions isn't added the project when you go the project with a full page reload, as you showed in your screenshot.

One issue I'm seeing that is also in `develop` is that the video extension will stay added to a project and the video will stay on when you navigate to another non-video sensing project without a full page refresh. Steps to repro:

1) Open project that uses the video sensing extension
2) Change the project ID in the URL manually to a project that doesn't use the video extension at all. Hit enter.
3) Observe the video is still on and the extension is still added

I'll make another issue for this since I'm not seeing an existing one (let me know if I missed it though). However, I wanted to mention it in case it comes up as you're taking a look at this round of changes as well.

**Comment by fsih** [comment]
@ktbee we have a bug for video staying on between projects here: https://github.com/LLK/scratch-gui/issues/1849

File: src/serialization/sb2.js
Comment (fsih) [inline src/serialization/sb2.js]: Nit, I don't think you need this if block

**Review by fsih — APPROVED** [review]
Looks great, thank you!

**Comment by ktbee** [comment]
@fsih good catch! I removed that if-statement you mentioned.

**Comment by ktbee** [comment]
Also @fsih , let me know if I should merge this or if we're waiting to add the "good to merge" label for another reason

**Comment by fsih** [comment]
@ktbee yes, when it's good to merge the writer of the change can merge the change. There was a Travis integration test failing, so I went into Travis and restarted it and it's succeeding now (unfortunately tests time out somewhat often :/) but now it's all passing so feel free to merge!

**Comment by fsih** [comment]
One thing to note is that once you merge the pull request, Travis tests kick off again, and if they don't pass, then the change doesn't actually get deployed to NPM. So if a test flakes again you might have to check back in and restart the Travis build :s

**Comment by BryceLTaylor** [comment]
This has caused the pen extension to not load: https://github.com/LLK/scratch-vm/issues/1676

**Pull request merged**
